tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/httpclient/RagondinHttpClient.java

```diff
@@ -0,0 +1,94 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+package com.linagora.tmail.mailet.rag.httpclient;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.linagora.tmail.mailet.rag.RagConfig;
+
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+public class RagondinHttpClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RagondinHttpClient.class);
+    private static final String API_KEY_HEADER = "Authorization";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private final HttpClient httpClient;
+    private final RagConfig configuration;
+    private final ObjectMapper objectMapper;
+
+
+    public RagondinHttpClient(RagConfig configuration) {
+        this.configuration = configuration;
+        this.objectMapper = new ObjectMapper().registerModule(new Jdk8Module());
+        this.httpClient = buildReactorNettyHttpClient(configuration);
+    }
+
+    Mono<Void> addDocument(Partition partition, DocumentId documentId, String textualContent, Map<String, String> metadata) throws JsonProcessingException {
+        String url = String.format("/indexer/partition/%s/file/%s", partition.partitionName(), documentId.asString());
```

Comment:
The partition name should be URL-encoded since the email address is somewhat arbitrary. Let's say the partition format is `{localPart}.twake.{domainName}`. The James `MailAddress` is parsed according to the definition from [RFC 821](https://datatracker.ietf.org/doc/html/rfc821) (pp. 30-31), where `localPart` could contain special characters, such as `/`. The resulting formatted URL would become invalid for the Ragondin API.
URL-encoding is a good practice as it prevents such edge cases.

Reply:
> The partition name should be URL-encoded since the email address is somewhat arbitrary.

While I tend to agree with this statement, the partition is coupled with other software of the TWP and thus it needs to be a coordinated choice at the workplace level, applied indeed for Twake Mail.

Cc @paultranvan